testnet/stacks-node/src/nakamoto_node/signer_coordinator.rs

```diff
@@ -254,46 +254,54 @@ impl SignerCoordinator {
         };
 
         let block_proposal_message = SignerMessageV0::BlockProposal(block_proposal);
-        debug!("Sending block proposal message to signers";
-            "signer_signature_hash" => %block.header.signer_signature_hash(),
-        );
-        Self::send_miners_message::<SignerMessageV0>(
-            &self.message_key,
-            sortdb,
-            election_sortition,
-            stackerdbs,
-            block_proposal_message,
-            MinerSlotID::BlockProposal,
-            self.is_mainnet,
-            &mut self.miners_session,
-            &election_sortition.consensus_hash,
-        )?;
-        counters.bump_naka_proposed_blocks();
 
-        #[cfg(test)]
-        {
-            info!(
-                "SignerCoordinator: sent block proposal to .miners, waiting for test signing channel"
+        loop {
+            debug!("Sending block proposal message to signers";
+                "signer_signature_hash" => %block.header.signer_signature_hash(),
             );
-            // In test mode, short-circuit waiting for the signers if the TEST_SIGNING
-            //  channel has been created. This allows integration tests for the stacks-node
-            //  independent of the stacks-signer.
-            if let Some(signatures) =
-                crate::tests::nakamoto_integrations::TestSigningChannel::get_signature()
+            Self::send_miners_message::<SignerMessageV0>(
+                &self.message_key,
+                sortdb,
+                election_sortition,
+                stackerdbs,
+                block_proposal_message.clone(),
+                MinerSlotID::BlockProposal,
+                self.is_mainnet,
+                &mut self.miners_session,
+                &election_sortition.consensus_hash,
+            )?;
+            counters.bump_naka_proposed_blocks();
```

Comment:
Should this be bumped on re-proposal?

Reply:
Yeah, I was unsure whether or not it made sense to bump this. What do you think?

Reply:
My thinking was yes, bump it, because we already have a separate count for mined blocks, which would only get bumped once, so let's bump this one each time we re-propose.

Reply:
EDIT: "... we already have a separate count for mined blocks and rejected blocks, ..."

Reply:
We can always go back and add a separate counter for novel proposed blocks, but if CI is passing with this change then it's fine with me.